modules/nw-udn-cr.adoc

```diff
@@ -34,32 +34,89 @@ spec:
-<7> Specifies the `subnets` used inside the OVN-Kubernetes network topology. If omitted, the OVN-Kubernetes network plugin assigns one, which is subject to change over time.
-<8> Specifies the IP address management system (IPAM). This field is optional and allowed when `topology` is `layer2`. The
-`subnets` field must be specified when this field is specified. The `Persistent` value specifies that workloads have persistent IP addresses. Assigned by the container network interface (CNI) and used by OVN-Kubernetes to program pod IP addresses. You must not change this for pod annotations.
+<5> The maximum transmission units (MTU) is an optional field. The default value is `1400`. The boundary for IPv4 is `574`, and for IPv6 it is `1280`.
+<6> For `Layer2` topology types the following specifies config details for the `subnet` field:
++
+* The subnets field is optional.
+* The subnets field is of type `string` and accepts standard CIDR formats for both IPv4 and IPv6.
+* The subnets field accepts one or two items. For two items, they must be of a different family. For example, subnets values of `10.100.0.0/16` and `2001:db8::/64`.
+* `Layer2` subnets may be omitted. If omitted, users must configure IP addresses for the pods. As a consequence, port security only prevents MAC spoofing.
+* The `Layer2` `subnets` field is mandatory when the `ipamLifecycle` field is specified.
+
++
+[NOTE]
+====
+There are two additional configurations that UDN uses, `spec.joinSubnets` and `spec.IPAMlifecycle`. It is not recommended to set these fields without explicit need and understanding of the OVN network topology.
+
+When omitted, the platform will set default values for the `joinSubnets` field of `100.65.0.0/16` for IPv4 and  `fd99::/64` for IPv6. If the default address values are used anywhere in the cluster's network you must override it by setting the `joinSubnets` field. If you choose to set this field, ensure it does not conflict with other subnets in the cluster such as the cluster subnet, the `default` network cluster subnet, and the masquerade subnet.
+
+The `joinSubnets` field configures the routing between different segments within the Layer 2 network. Dual-stack clusters may set 2 subnets, one for each IP family, otherwise only 1 subnet is allowed. This field is only allowed for the `Primary` network.
+
+The `IPAMLifecycle` field configures the IP address management system (IPAM). This field is allowed when `topology` is `layer2`. The `subnets` field must be specified when this field is specified. The `Persistent` value specifies that workloads have persistent IP addresses. These are assigned by the container network interface (CNI) and used by OVN-Kubernetes to program pod IP addresses. You must not change this for pod annotations.
```

Comment:
> The `Persistent` value specifies that workloads have persistent IP addresses.

This is a bit misleading. Only the workloads that request a persistent IP will have one - not all IPs will be persistent. For instance, **all** virt workloads attached to this network will request a persistent IP on this particular attachment *if* the network has this knob enabled.

And IMHO we should indicate when this is supposed to be used; this knob is recommended to be used if you are attaching virt-workloads to this network.

Reply:
The point is that not everything connected to a network with this knob enabled will have persistent IPs; only the workloads requesting it will. I.e. while it is a network wide setting, it does **not** apply to all attachments to the network

> Setting a value of `Persistent` ensures that your virtual workloads have persistent IP addresses.

I would clarify it further with: "Setting a value of `Persistent` ensures that your virtual machine workloads have persistent IP addresses across reboots, and migration."

Reply:
done